template/java8/model/src/main/java/com/openfaas/model/Request.java

```diff
@@ -17,23 +20,39 @@
     private String queryRaw;
     private String pathRaw;
     private Map<String, String> path;
+    private InputStream inputStream;
 
-    public Request(String body, Map<String, String> headers) {
-        this.body = body;
+    public Request(Map<String, String> headers) {
         this.headers = headers;
     }
     
-    public Request(String body, Map<String, String> headers,String queryRaw, String path) {
-        this.body = body;
+    public Request(Map<String, String> headers, String queryRaw, String path, InputStream inputStream) {
         this.headers = headers;
         this.queryRaw = queryRaw;
-        this.queryParameters = this.parseQueryParameters();
+	    this.inputStream = inputStream;
+	    this.queryParameters = this.parseQueryParameters();
         this.pathRaw = path;
         this.path = this.parsePathParameters();
     }
 
     public String getBody() {
-        return this.body;
+	    if (this.body != null) {
+	    	return this.body;
+	    }
+	    try {
+		    ByteArrayOutputStream result = new ByteArrayOutputStream();
+		    byte[] buffer = new byte[1024];
+		    int length;
+		    while ((length = inputStream.read(buffer)) != -1) {
+			    result.write(buffer, 0, length);
+		    }
+		    // StandardCharsets.UTF_8.name() > JDK 7
+		    this.body = result.toString("UTF-8");
+		    return this.body;
+	    } catch (IOException e) {
+	    	// this method should be throwing the exception, but that would break with current behaviour
+		    return "";
```

Comment:
Shall we perhaps do *something* with it?

Reply:
Could perhaps throw a runtime exception. If an exception is added to the method signature (which would be the best solution) that will break with current usages of this method.